### PR TITLE
feat(cli): add `source-smoke-test` CLI

### PR DESCRIPTION
**Pull request by aaronsteers**
## Summary

Extracts the smoke-test source from [PR #969](https://github.com/airbytehq/PyAirbyte/pull/969) into its own dedicated submodule at `airbyte/cli/smoke_test_source/`, so it can be reviewed and merged independently from the broader universal connector work.

The new `source-smoke-test` CLI entrypoint is registered in `pyproject.toml`.

Related: [#995](https://github.com/airbytehq/PyAirbyte/issues/995) (ops repo test harness delivery), [#969](https://github.com/airbytehq/PyAirbyte/pull/969) (parent universal connector PR)

**Structure:**
- `airbyte/cli/smoke_test_source/_scenarios.py` — 15 predefined test scenarios (basic types, nulls, naming edge cases, wide tables, unicode, large batches, etc.)
- `airbyte/cli/smoke_test_source/source.py` — `SourceSmokeTest` class (extends CDK `Source`)
- `airbyte/cli/smoke_test_source/run.py` — thin CLI entry point

**Structural change — `cli.py` → `cli/` package:**

Adding the `smoke_test_source/` submodule required converting `airbyte/cli.py` (a module) into `airbyte/cli/` (a package), which would shadow the original module. To resolve this:
- `airbyte/cli.py` → `airbyte/cli/pyab.py` (preserves git history via `git mv`)
- `airbyte/cli/__init__.py` re-exports `cli` for backwards compatibility
- Entry points updated to `airbyte.cli.pyab:cli`
- `docs/generate.py` updated to reference the new path

**Other fixes included:**
- Type validation for `scenario_filter` and `custom_scenarios` config fields (guards against `None`/wrong types)
- `_get_connector_name()` fix: uses `maxsplit=1` and `rsplit()` for correct Docker image name parsing

## Review & Testing Checklist for Human

- [ ] **Verify `pyairbyte --help` and `pyab --help` still work** — the CLI module was converted from `airbyte/cli.py` to a package at `airbyte/cli/`. Entry points now resolve via `airbyte.cli.pyab:cli`. Confirm both commands work after `pip install -e .`
- [ ] **Verify `source-smoke-test spec` runs** — install in dev mode and run `source-smoke-test spec` to confirm the entrypoint resolves and outputs a valid connector spec JSON
- [ ] **Spot-check `docs/generate.py`** — the pdoc path changed from `airbyte/cli.py` to `airbyte/cli/pyab.py`. Run `poe docs-generate` to confirm docs still build correctly
- [ ] **Review scenario data for correctness** — spot-check a few predefined scenarios (e.g., `null_handling`, `column_naming_edge_cases`, `large_batch_stream`) to confirm the JSON schemas match the inline records

### Notes
- ~840 lines added with no tests. The scenarios and source logic (`check`, `discover`, `read`, `_get_all_scenarios`) are untested. Determine if tests should be added before merge or tracked as follow-up.
- The scenario data uses `dict[str, Any]` throughout (carried over from PR #969). A future improvement could type these with Pydantic models or dataclasses.
- Business logic lives under `cli/` per explicit request, though the project convention prefers CLI modules as thin wrappers. This is an intentional placement decision.
- No Dockerfile is included — containerization will be handled by the companion monorepo scaffold PR.

## Summary by CodeRabbit

* **New Features**
  * Added a Smoke Test Source that emits synthetic datasets across diverse predefined scenarios (nested data, nulls, edge-case names, wide tables, high-volume streams).
  * Added a CLI command to run predefined or custom scenarios with configurable large-batch generation and selective fast/slow stream options.

* **Bug Fixes**
  * Fixed Docker image name parsing so tags and registries are handled correctly.

* **Documentation**
  * Added module-level docs and public exports to surface the smoke test source in tooling and docs.

Link to Devin session: https://app.devin.ai/sessions/9c72389579884c06bf18cef11c4550e8
Requested by: @aaronsteers

> [!NOTE]
> **Auto-merge may have been disabled. Please check the PR status to confirm.**

### CODE REVIEW

File: airbyte/cli/smoke_test_source/source.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+                    "large_batch_record_count": {
+                        "type": "integer",
+                        "title": "Large Batch Record Count",
+                        "description": (
+                            "Number of records to generate for "
+                            "the large_batch_stream scenario. "
+                            "Set to 0 to skip this stream."
+                        ),
+                        "default": 1000,
+                    },
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte/cli/smoke_test_source/source.py:142]: Devin ☝️

File: airbyte/cli/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+"""CLI modules for PyAirbyte."""
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte/cli/__init__.py:2]: Devin ☝️

[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte/cli/__init__.py:2]: Move the existing /cli.py package to a sibling /cli/pyab.py and register both pyab and pyairbyte from there.

File: airbyte/cli/smoke_test_source/source.py [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+           ``all_slow_streams`` (default false) enables
+           high-volume scenarios.
+        2. ``scenario_filter`` names are unioned with the boolean sets.
+        3. Custom scenarios are always included.
+        4. The final list is deduplicated by name.
+        """
+        include_default = config.get("all_fast_streams", True)
+        include_high_volume = config.get("all_slow_streams", False)
+        scenario_filter: list[str] = config.get("scenario_filter", [])
+        explicit_names: set[str] = set(scenario_filter)
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte/cli/smoke_test_source/source.py]: Devin ☝️ wdyt?

File: airbyte/cli/__main__.py
Comment (aaronsteers) [inline airbyte/cli/__main__.py:1]: This file probably shouldn't exist. Bad smell.

File: airbyte/cli/__main__.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+"""Allow running the CLI via python -m airbyte.cli."""
```
Comment (aaronsteers) [inline airbyte/cli/__main__.py]: We have proper entrypoints. The -m syntax is pointless and archaic.

File: airbyte/cli/smoke_test_source/source.py [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+            if not scenario.get("name"):
+                return AirbyteConnectionStatus(
+                    status=Status.FAILED,
+                    message=f"Custom scenario at index {i} is missing 'name'.",
+                )
+            if not scenario.get("json_schema"):
+                return AirbyteConnectionStatus(
+                    status=Status.FAILED,
+                    message=(f"Custom scenario '{scenario['name']}' is missing 'json_schema'."),
+                )
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte/cli/smoke_test_source/source.py]: Devin, wdyt? ☝️
